### PR TITLE
batch requests client checks for ssl config and add test

### LINKED ISSUE #1456: TykBatchRequest should obey proxy_ssl_insecure_skip_verify
**Do you want to request a *feature* or report a *bug*?**

feature/bug

**What is the current behavior?**

`TykBatchRequest` can't make calls to self-signed CA or mis-matched hostname backends

```
[JSVM]: Request failed: Get https://mybackend.mydomain.com/whatever: x509: certificate is valid for xxxxx, not mybackend.mydomain.com"
```

**What is the expected behavior?**

There should be an option/param/flag for `TykBatchRequest` OR it should just obey what the gateway's `proxy_ssl_insecure_skip_verify` setting says

**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**

2.5.x

@see https://github.com/TykTechnologies/tyk/issues/1006

### PR DESCRIPTION
**Pull request by joshblakeley**
Should fix #1456 

Not sure if using a new client for each request is good practice? But should be ok as we close each request body.

### CODE REVIEW

**Comment by joshblakeley** [comment]
Any improvements on this @buger ?

File: batch_requests.go [hunk 1]
```diff
@@ -40,7 +41,18 @@ type BatchRequestHandler struct {
 
 // doRequest will make the same request but return a BatchReplyUnit
 func (b *BatchRequestHandler) doRequest(req *http.Request, relURL string) BatchReplyUnit {
-	resp, err := http.DefaultClient.Do(req)
+	tr := &http.Transport{TLSClientConfig: &tls.Config{}}
+	if cert := getUpstreamCertificate(req.Host, b.API); cert != nil {
+		tr.TLSClientConfig.Certificates = []tls.Certificate{*cert}
+	}
+
+	if config.Global.ProxySSLInsecureSkipVerify {
```
Comment (buger) [inline batch_requests.go]: Can be replaced by `tr.TLSClientConfig.InsecureSkipVerify = config.Global.ProxySSLInsecureSkipVerify`

**Comment by buger** [comment]
This is not really a fair way to test things like this :) 

The issue initially tells about JSVM api calls. JSVM implementation use `ManualBatchRequest` function instead of `BatchRequest`, and main idea here that JSVM batch function works with absolute urls, while our internal batch API works with relative URLs. 

So the valid way to test this feature would be creating test similar to `TestVirtualEndpoint`, but in this case, you create test https server using `https://golang.org/pkg/net/http/httptest/#Server.StartTLS` and inside virtual endpoint call this endpoint using `TykBatchRequest`. 

Also, note that good test should test both success and failure. So you first run it without with `config.Global.ProxySSLInsecureSkipVerify` false, and next enable it and it should pass. 

Note that functionality itself looks nice, and for sure this maybe function not that big to require full blown test but treat it as an exercise :)

**Comment by joshblakeley** [comment]
Test improved and ready for review @buger

**Comment by buger** [comment]
The test looks good! But you can remove SSL configuration of tyk itself, since it is not used here at all, it will help simplify test a lot.

**Comment by buger** [comment]
@joshblakeley I simplified test as suggested check the diff https://github.com/TykTechnologies/tyk/pull/1471/commits/05b5649760513c80830271bb8bf46cb6990c58cd

Sorry for intervention, just want get it included into patch release

**Pull request merged**